firestore/runtime/src/main/java/io/quarkiverse/googlecloudservices/firestore/runtime/config/FirestoreRetryConfiguration.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkiverse.googlecloudservices.firestore.runtime.config;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Controls the retry settings for the request retries of the Firestore service.
+ */
+@ConfigRoot(name = "google.cloud.firestore.retry", phase = ConfigPhase.RUN_TIME)
+public class FirestoreRetryConfiguration {
```

Comment:
`FirestoreRetryConfiguration`  should be used as a field inside `FirestoreConfiguration` not as a separate config root.

You can even nested the `FirestoreRetryConfiguration`  class inside the  `FirestoreConfiguration` class, it's not mandatory but it somplifies things a bit.

See for example https://github.com/quarkusio/quarkus/blob/main/extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java

Reply:
And it should be an optional configuration.

Reply:
Thanks, I have changed it. I agree, this is now much better.